src/plugins/janus_sip.c

```diff
@@ -7392,6 +7401,52 @@ gpointer janus_sip_sofia_thread(gpointer user_data) {
 	return NULL;
 }
 
+/* Check peer RTP has RFC2833 and push event */
+static void janus_sip_check_rfc2833(janus_sip_session *session, char *buffer, int len) {
+	janus_rtp_header *rtp_header = (janus_rtp_header *)buffer;
+	if(rtp_header->type != JANUS_RTP_RFC2833_DEFAULT_PAYLOAD_TYPE)
+		return;
+	int plen = 0;
+	char *payload_buffer = janus_rtp_payload(buffer, len, &plen);
+	if(plen < sizeof(janus_rtp_rfc2833_payload))
```

Comment:
The compiler compains about the different signedness, here, since plen is a signed integer, while sizeof returns an unsigned (`size_t`). I guess a possible fix could be something like this

	if(plen < 0 || (size_t)plen < sizeof(janus_rtp_rfc2833_payload))